docs/source/tutorials_source/package/tutorial_moco_memory_bank.py

```diff
@@ -264,58 +291,69 @@ def configure_optimizers(self):
-    def __init__(self, model):
+    def __init__(self, backbone):
         super().__init__()
-        # create a moco based on ResNet
-        self.resnet_moco = model
+        # use the pretrained ResNet backbone
+        self.backbone = backbone
 
-        # freeze the layers of moco
-        for p in self.resnet_moco.parameters():  # reset requires_grad
-            p.requires_grad = False
+        # freeze the backbone
+        deactivate_requires_grad(backbone)
 
-        # we create a linear layer for our downstream classification
-        # model
+        # create a linear layer for our downstream classification model
         self.fc = nn.Linear(512, 10)
 
-        self.accuracy = pl.metrics.Accuracy()
+        self.criterion = nn.CrossEntropyLoss()
 
     def forward(self, x):
-        with torch.no_grad():
-            y_hat = self.resnet_moco.backbone(x).squeeze()
-            y_hat = nn.functional.normalize(y_hat, dim=1)
+        y_hat = self.backbone(x).flatten(start_dim=1)
         y_hat = self.fc(y_hat)
         return y_hat
 
-    # We provide a helper method to log weights in tensorboard
-    # which is useful for debugging.
-    def custom_histogram_weights(self):
-        for name, params in self.named_parameters():
-            self.logger.experiment.add_histogram(
-                name, params, self.current_epoch)
-
     def training_step(self, batch, batch_idx):
         x, y, _ = batch
         y_hat = self.forward(x)
-        loss = nn.functional.cross_entropy(y_hat, y)
-        self.log('train_loss_fc', loss)
+        loss = self.criterion(y_hat, y)
+        self.log("train_loss_fc", loss)
         return loss
 
     def training_epoch_end(self, outputs):
         self.custom_histogram_weights()
 
+    # We provide a helper method to log weights in tensorboard
+    # which is useful for debugging.
+    def custom_histogram_weights(self):
+        for name, params in self.named_parameters():
+            self.logger.experiment.add_histogram(
+                name, params, self.current_epoch
+            )
+
     def validation_step(self, batch, batch_idx):
         x, y, _ = batch
         y_hat = self.forward(x)
         y_hat = torch.nn.functional.softmax(y_hat, dim=1)
-        self.accuracy(y_hat, y)
-        self.log('val_acc', self.accuracy.compute(),
-                 on_epoch=True, prog_bar=True)
+
+        # calculate number of correct predictions
+        _, predicted = torch.max(y_hat, 1)
+        num = predicted.shape[0]
+        correct = (predicted == y).float().sum()
+        return num, correct
+
+    def validation_epoch_end(self, outputs):
+        # calculate and log top1 accuracy
+        if outputs:
+            total_num = torch.Tensor([0],).to(self.device)
+            total_correct = torch.Tensor([0.0],).to(self.device)
+            for num, correct in outputs:
+                total_num += num
+                total_correct += correct
+            acc = total_correct.item() / total_num.item()
```

Comment:
What is the reason here to use a GPU tensor for two single scalars?

Wouldn't it be easier to compute it in plain python?
```python
total_num =0
total_correct = 0
for num, correct in outputs:
   total_num += num.item()
   total_correct += correct.item()
acc = total_correct / total_item
```

Reply:
Yes you are correct! I updated the pr accordingly.